string.c

```diff
@@ -10450,33 +10450,59 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
 /**********************************************************************
  * Document-class: Symbol
  *
- *  <code>Symbol</code> objects represent names and some strings
- *  inside the Ruby
- *  interpreter. They are generated using the <code>:name</code> and
- *  <code>:"string"</code> literals
- *  syntax, and by the various <code>to_sym</code> methods. The same
- *  <code>Symbol</code> object will be created for a given name or string
- *  for the duration of a program's execution, regardless of the context
- *  or meaning of that name. Thus if <code>Fred</code> is a constant in
- *  one context, a method in another, and a class in a third, the
- *  <code>Symbol</code> <code>:Fred</code> will be the same object in
- *  all three contexts.
- *
- *     module One
- *       class Fred
+ *  <code>Symbol</code> objects are often used like <code>String</code>
+ *  objects to represent names.
+ *
+ *  Symbols are generated using either of the following literal
+ *  syntaxes:
+ *
+ *  * <code>:name</code>
+ *  * <code>:"name with spaces"</code>
+ *
+ *  You can also convert some Ruby objects to symbols using common
+ *  <code>to_sym</code> conversion methods:
+ *
+ *     "some name".to_sym
+ *     => :"some name"
+ *
+ *  Symbols are often used instead of Strings because the same
+ *  <code>Symbol</code> object will be used in memory for a given
+ *  name or string for the duration of a program's execution.
+ *
+ *     class StringExample
+ *       def one
+ *         "important name"
+ *       end
+ *
+ *       def two
+ *         "important name"
  *       end
- *       $f1 = :Fred
- *     end
- *     module Two
- *       Fred = 1
- *       $f2 = :Fred
  *     end
- *     def Fred()
+ *
+ *     class SymbolExample
+ *       def one
+ *         :"important name"
```

Comment:
The `:"string"` syntax is much less common than the `:string` form. When comparing `"important name"` and `:"important name"` in the example, it's not clear that the symbol syntax is often shorter and might be preferred for readability, and there's a risk that the quotes may cause confusion as there's still a string there.

I think comparing `'name'` and `:name` is clearer. As symbols don't allow spaces, it does mean changing the 'important name' example, which is a pity. 

Would it be better to change the example to:

```ruby
class StringExample
  def one
    "example"
  end

  def two
    "example"
  end
end

class SymbolExample
  def one
    :example
  end

  def two
    :example
  end
end
```

?

Reply:
Symbols allow spaces?

```
pry(main)> :'foo bar'
=> :"foo bar"
pry(main)> "foo bar".to_sym
=> :"foo bar"
```

Reply:
Sorry, yes, they do, but they require the confusing - to Ruby learners - mixture of `:` _and_ `"`. The shorter `:symbol` form is much more commonly used, and I wanted to make sure we were comparing `"string"` and `:symbol` in the way the Ruby learner is most likely to encounter them first.

Reply:
That makes sense.

Reply:
@TheTeaNerd I sort of disagree. A lot of the confusion I've encountered from beginners stems from looking at symbols as some sort of weird entity that seems to be used in Ruby for some specific but indecipherable (at first) reason. They're just strings, and giving an example of their permissive literal syntax is a great way to help beginners understand that there's nothing magical about the `:` + `word` syntax. It's just a shorthand.

This is an introduction to the Symbol class, not an introduction to Ruby or even the String class. I would understand your objection to the symbol-with-space example if we were doing a quick Ruby intro but this is one of the biggest issues with Ruby: normal behavior isn't documented well. By showing that symbols can have spaces and any character strings can have, we're opening the door for people to use interpolation to create symbols. 

Moreover, the shorthand symbol literal syntax is the firs thing shown at the very beginning of the class documentation.